src/core/clifford.jl

```diff
@@ -0,0 +1,397 @@
+end
+
+function unsafe_get_pauli(gate::Identity, num_qubits::Integer,
+    imaginary_exponent::GFInt, negative_exponent::GFInt)::PauliGroupElement
+    
+    u = zero_matrix(GF(2), 2*num_qubits, 1)
+    return PauliGroupElement(u, imaginary_exponent, negative_exponent)
+end
+
+function unsafe_get_pauli(gate::SigmaX, num_qubits::Integer,
+    imaginary_exponent::GFInt, negative_exponent::GFInt)::PauliGroupElement
+    
+    target_qubit = get_connected_qubits(gate)[1]
+    u = zero_matrix(GF(2), 2*num_qubits, 1)
+    u[num_qubits+target_qubit, 1] = 1
+    return PauliGroupElement(u, imaginary_exponent, negative_exponent)
+end
+
+function unsafe_get_pauli(gate::SigmaY, num_qubits::Integer,
+        imaginary_exponent::GFInt, negative_exponent::GFInt)::PauliGroupElement
+    target_qubit = get_connected_qubits(gate)[1]
+    u = zero_matrix(GF(2), 2*num_qubits, 1)
+    u[target_qubit, 1] = 1
+    u[num_qubits+target_qubit, 1] = 1
+    negative_exponent += imaginary_exponent+1
+    imaginary_exponent += 1
+    return PauliGroupElement(u, imaginary_exponent, negative_exponent)
+end
+
+function unsafe_get_pauli(gate::SigmaZ, num_qubits::Integer,
+    imaginary_exponent::GFInt, negative_exponent::GFInt)::PauliGroupElement
+    
+    target_qubit = get_connected_qubits(gate)[1]
+    u = zero_matrix(GF(2), 2*num_qubits, 1)
+    u[target_qubit, 1] = 1
+    return PauliGroupElement(u, imaginary_exponent, negative_exponent)
+end
+
+"""
+    Base.:*(p1::PauliGroupElement, p2::PauliGroupElement)::PauliGroupElement
+
+Returns the product of two `PauliGroupElement` objects.
+
+The `PauliGroupElement` objects must be associated with the same number of qubits.
+
+# Examples
+```jldoctest
+julia> pauli_z = get_pauli(sigma_z(1), 1)
+Pauli Group Element:
+1.0*Z(1)
+
+
+
+julia> pauli_y = get_pauli(sigma_y(1), 1)
+Pauli Group Element:
+1.0*Y(1)
+
+
+
+julia> pauli_z*pauli_y
+Pauli Group Element:
+-1.0im*X(1)
+
+
+
+```
+"""
+function Base.:*(p1::PauliGroupElement, p2::PauliGroupElement)::PauliGroupElement
+
+    new_delta = p1.delta+p2.delta
+    if nrows(p1.u) != nrows(p2.u)
+        throw(ErrorException("the Pauli group elements must be associated with the "
+        *"same number of qubits"))
+    end
+    new_u = p1.u+p2.u
+    twice_num_qubits = nrows(new_u)
+    num_qubits = Int(twice_num_qubits/2)
+    capital_u = zero_matrix(GF(2), twice_num_qubits, twice_num_qubits)
+    capital_u[1:num_qubits, num_qubits+1:twice_num_qubits] =
+        identity_matrix(GF(2), num_qubits)
+    new_epsilon = p1.epsilon+p2.epsilon+p1.delta*p2.delta+transpose(p2.u)*capital_u*p1.u
+    return PauliGroupElement(new_u, new_delta, new_epsilon[1,1])
+end
+
+function Base.:(==)(lhs::PauliGroupElement, rhs::PauliGroupElement)::Bool
+    if lhs.delta != rhs.delta
+        return false
+    elseif lhs.epsilon != rhs.epsilon
+        return false
+    elseif lhs.u != rhs.u
+        return false
+    else
+        return true
+    end
+end
+
+struct DisplayablePauli
```

Comment:
I think a better name could be `PauliCircuit`

Reply:
The `struct` has been renamed.